docs/plugins/mcp.mdx

```diff
@@ -0,0 +1,410 @@
+      ],
+    })
+    return collection
+  },
+  // ... other options
+})
+```
+
+## Custom Prompts
+
+Prompts allow LLMs to generate structured messages for specific tasks. Each prompt defines a schema for arguments and returns formatted messages:
+
+```ts
+prompts: [
+  {
+    name: 'reviewContent',
+    title: 'Content Review Prompt',
+    description: 'Creates a prompt for reviewing content quality',
+    argsSchema: {
+      content: z.string().describe('The content to review'),
+      criteria: z.array(z.string()).describe('Review criteria'),
+    },
+    handler: ({ content, criteria }) => ({
+      messages: [
+        {
+          content: {
+            type: 'text',
+            text: `Please review this content based on the following criteria: ${criteria.join(', ')}\n\nContent: ${content}`,
+          },
+          role: 'user',
+        },
+      ],
+    }),
+  },
+]
+```
+
+## Custom Resources
+
+Resources provide access to data or content that LLMs can read. They can be static or dynamic with parameterized URIs:
+
+```ts
+resources: [
+  // Static resource
+  {
+    name: 'guidelines',
+    title: 'Content Guidelines',
+    description: 'Company content creation guidelines',
+    uri: 'guidelines://company',
+    mimeType: 'text/markdown',
+    handler: (uri) => ({
+      contents: [
+        {
+          uri: uri.href,
+          text: '# Content Guidelines\n\n1. Keep it concise\n2. Use clear language',
+        },
+      ],
+    }),
+  },
+
+  // Dynamic resource with template
+  {
+    name: 'userProfile',
+    title: 'User Profile',
+    description: 'Access user profile information',
+    uri: new ResourceTemplate('users://profile/{userId}', { list: undefined }),
+    mimeType: 'application/json',
+    handler: async (uri, { userId }) => {
+      // Fetch user data from your system
+      const userData = await getUserById(userId)
+      return {
+        contents: [
+          {
+            uri: uri.href,
+            text: JSON.stringify(userData, null, 2),
+          },
+        ],
+      }
+    },
+  },
+]
+```
+
+### Data Override Example
+
+Use override functions to modify data before it's processed:
+
+```ts
+collections: {
+  posts: {
+    enabled: true,
+    override: (original, req) => {
+      // Log the operation
+      req.payload.logger.info('MCP operation on posts:', original)
+
+      // Sanitize or validate data
+      const sanitized = {
+        ...original,
+        title: sanitizeTitle(original.title),
+        publishedAt: original.publishedAt || new Date().toISOString(),
+      }
+
+      return sanitized
+    },
+  },
+}
+```
+
+Use overrideResponse to modify response data:
+
+```ts
+collections: {
+  posts: {
+    enabled: true,
+    overrideResponse: (response, doc, req) => {
+      response.content.push({
+        type: 'text',
+        text: 'Additional response for Posts!',
+      })
+      return response
+    },
+  },
+}
+```
+
+### Custom Tool with Database Access
+
```

Comment:
Could use a general 1 or sentences about why you need it and when to reach for it.